vault/resource_ssh_secret_backend_ca.go

```diff
@@ -0,0 +1,140 @@
+package vault
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+	"log"
+	"strings"
+)
+
+func sshSecretBackendCAResource() *schema.Resource {
+	return &schema.Resource{
+		Create: sshSecretBackendCACreate,
+		Read:   sshSecretBackendCARead,
+		Delete: sshSecretBackendCADelete,
+		Exists: sshSecretBackendCAExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "ssh",
+				ForceNew:    true,
+				Description: "The path of the SSH Secret Backend where the CA should be configured",
+				// standardise on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+			"generate_signing_key": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				ForceNew:    true,
+				Description: "Whether Vault should generate the signing key pair internally.",
+			},
+			"private_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Computed:    true,
+				Description: "Private key part the SSH CA key pair; required if generate_signing_key is false.",
+			},
+			"public_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Public key part the SSH CA key pair; required if generate_signing_key is false.",
+			},
+		},
+	}
+}
+
+func sshSecretBackendCACreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+	generateSigningKey := d.Get("generate_signing_key").(bool)
+
+	data := map[string]interface{}{
+		"generate_signing_key": generateSigningKey,
```

Comment:
I was looking at the code this feeds into [here](https://github.com/hashicorp/vault/blob/df18871704fe869e9be45b542a6b1eb2fe46c293/builtin/logical/ssh/path_config_ca.go#L154-L191) and it actually behaves differently based on whether the `generate_signing_key` flag was _explicitly_ set to false or simply omitted. So, I think that for this code to behave as expected, we would probably want to only include this flag _if it were set_.

That would mean that above, we'd want to do: 
```
generateSigningKey, ok := d.Get("generate_signing_key").(bool)
if ok {
    data["generate_signing_key"] = generateSigningKey
}
```